src/Medidata.ZipkinTracer.Core.Collector/SpanProcessor.cs

```diff
@@ -53,71 +57,62 @@ public virtual void Stop()
-                || (subsequentPollCount > MAX_NUMBER_OF_POLLS))
-            {
-                var entries = logEntries;
-                logEntries = new List<LogEntry>();
-                subsequentPollCount = 0;
-                Log(clientProvider, entries);
-            }
-        }
 
-        internal void Log(IClientProvider client, List<LogEntry> logEntries)
-        {
-            try
-            {
-                clientProvider.Log(logEntries);
-                retries = 0;
-            }
-            catch (TException)
+            if ((serializableSpans.Count() >= maxBatchSize)
+                || (serializableSpans.Any() && spanProcessorTaskFactory.IsTaskCancelled())
+                || (subsequentPollCount > MAX_NUMBER_OF_POLLS))
             {
-                if ( retries < 3 )
+                try
                 {
-                    retries++;
-                    Log(client, logEntries);
+                    if (serializableSpans.Any()) 
+                        SendSpansToZipkin(JsonConvert.SerializeObject(serializableSpans.ToList()));
                 }
-                else
+                catch (WebException ex)
                 {
-                    throw;
+                    logger.Error("Failed to send to zipking with error: " + ex.Message);
                 }
             }
         }
 
-        private LogEntry Create(Span span)
+        public virtual void SendSpansToZipkin(string requestBody)
         {
-            var spanAsString = Convert.ToBase64String(ConvertSpanToBytes(span));
-            return new LogEntry()
+            using (var client = new WebClient())
             {
-                Category = "zipkin",
-                Message = spanAsString
-            };
-        }
+                const string tracerPath = "/api/v1/spans";
 
-        private byte[] ConvertSpanToBytes(Span span)
-        {
-            var buf = new MemoryStream();
-            TProtocol protocol = protocolFactory.GetProtocol(new TStreamTransport(buf, buf));
-            span.Write(protocol);
-            return buf.ToArray();
+                try
+                {
+                    client.BaseAddress = uri.ToString();
+                    client.UploadString(tracerPath, "POST", requestBody);
+                }
+                catch (WebException ex)
+                {
+                    var response = ex.Response as HttpWebResponse;
+                    if ((response != null))
+                    {
+                        // read all response info
+                        var responseStream = response.GetResponseStream();
+                        if (responseStream != null)
+                        {
+                            var responseString = new StreamReader(responseStream).ReadToEnd();
+                            logger.Error("Sending Spans to Failed with error: " + ex.Message + " and message:" + responseString);
+                        }
```

Comment:
same with my line 85. Please throw exception error. (please see above for reason). Thanks.